Fix sqlite in docker image

ComfortablyCoding opened the pull request:
<!--

Heya! Thanks for opening a Pull Request! If your PR is implementing a new feature or fix for Directus, please make sure your PR adheres to the following requirements:

- The PR closes an Issue (not Discussion)
- Tests are added/updated and are passing locally if applicable
- Documentation was added/updated if applicable

Please make sure to "Link" the issue you're closing. Without a Linked issue, this PR won't be accepted. See https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue for more information.

-->

## Scope

What's changed:

- The necessary `py3-setuptools` package has been added to the docker image build process.

## Potential Risks / Drawbacks

- None

## Review Notes / Questions

- `py3-setuptools` is now required as a separate entity due to `distutils` (deprecated in favour of `py3-setuptools`) being removed from `python3` starting with `3.12.X`. When upgrading from node `node v18` to support `node v22` the `python3` version was changed from `3.11.X` to `3.12.X` in the underlying alpine image. `distutils`/`py3-setuptools` is utilized by the sqlite3 build process and results in a build error if not present
- Tested it to work with at least `linux/arm64` and `linux/amd64` (thanks @br41nslug!)

---

Fixes #24727
Fixes #24612

ComfortablyCoding committed:
fix docker image build

ComfortablyCoding committed:
add changeset

ComfortablyCoding committed:
add missing EOF

br41nslug reviewed (approved):
Can confirm this fixes the sqlite3 issue for the `linux/amd64` build target on my end. Would love for infra to confirm this works on their end too :D

paescuj committed:
Simplify RUN statement

paescuj committed:
Drop superfluous symlink

This is not necessary anymore since it's now provided by the python
package itself, see https://pkgs.alpinelinux.org/contents?name=python3&repo=main&branch=v3.21&arch=x86_64

paescuj committed:
Fail whole build if cmd in install block fails

paescuj committed:
Clarify scope of change

Other db vendor are unaffected

paescuj commented:
This was actually a bad coincidence of two things:

- In this week's release [v11.5.0](https://github.com/directus/directus/releases/tag/v11.5.0), the Node.js version was updated to 22.14.0 in which the Node API was increased [^2]. This in turn caused an issue with `prebuild-install` which couldn't detect the API version.
  See our build log [^3]:
  ```console
  245.0 .../sqlite3@5.1.7/node_modules/sqlite3 install: prebuild-install warn This package does not support N-API version undefined
  245.2 .../sqlite3@5.1.7/node_modules/sqlite3 install: prebuild-install warn install No prebuilt binaries found (target=undefined runtime=napi arch=x64 libc=musl platform=linux)
  ```
  Therefore, the prebuild binary couldn't be fetched.
  This has been fixed in the meantime, see also https://github.com/nodejs/node/issues/57118#issuecomment-2690138016

- As already mentioned by @ComfortablyCoding, when we switched the base image to `node:22-alpine`, Python was updated to 3.12 in which `distutils` has been dropped [^1]. This resp. its successor `setuptools` is however required to build the node bindings in `sqlite3` from source. We haven't noticed this problem so far, as the prebuild binary was fetched in the past (no build from source necessary).

In the meantime, the prebuilts are used again for both platforms. To make sure that the fix actually works, I tested it with the `npm_config_build_from_source=true` flag for both platforms.

Note: Although the build dependencies are therefore no longer strictly necessary, we are now keeping them in **for both platforms** to be on the safe side in case there should be any problems with prebuilts again for whatever reason. Also for `arm64` it used to be required for `isolated-vm` as well, and probably still is as of today.

[^1]: https://docs.python.org/3.12/whatsnew/3.12.html#summary-release-highlights
[^2]: https://github.com/nodejs/node/pull/55676
[^3]: https://github.com/directus/directus/actions/runs/13550774895/job/37873610619#step:9:1834

paescuj reviewed (approved):
Thank you, @ComfortablyCoding ❤️
